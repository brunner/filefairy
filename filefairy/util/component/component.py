#!/usr/bin/env python3
# -*- coding: utf-8 -*-


def anchor(url, content):
    return '<a href="{}">{}</a>'.format(url, content)


def card(href='', title='', info='', table=None, ts='', success='', danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }


def span(classes, text):
    return '<span class="{}">{}</span>'.format(' '.join(classes), text)


def table(clazz='border mt-3', hcols=None, bcols=None, head=None, body=None):
    return {
        'clazz': clazz,
        'hcols': hcols,
        'bcols': bcols,
        'head': head,
        'body': body
    }
