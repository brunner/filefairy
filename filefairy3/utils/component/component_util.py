#!/usr/bin/env python


def card(href='',
         title='',
         info='',
         table=[],
         ts='',
         success='',
         danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }
