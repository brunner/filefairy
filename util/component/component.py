#!/usr/bin/env python3
# -*- coding: utf-8 -*-


def anchor(url, content):
    return '<a href="{}">{}</a>'.format(url, content)


def bold(text):
    return span(['font-weight-bold'], text)


def card(href='',
         title='',
         info='',
         code='',
         table=None,
         ts='',
         success='',
         danger=''):
    return {
        'href': href,
        'title': title,
        'info': info,
        'code': code,
        'table': table,
        'ts': ts,
        'success': success,
        'danger': danger
    }


def cell(col=None, content=''):
    return {'col': col, 'content': content}


def col(clazz='', colspan=''):
    return {'clazz': clazz, 'colspan': colspan}


def show(trigger, text):
    s = '<a data-toggle="collapse" href="#{0}" role="button" ' + \
        'aria-expanded="false" aria-controls="{0}">{1}</a>'
    return s.format(trigger, text)


def replace(text, replace):
    s = '<a class="repl-toggler" href="#" role="button" data-repl="{}">{}</a>'
    return s.format(replace, text)


def secondary(text):
    return span(['text-secondary'], text)


def span(classes, text):
    return '<span class="{}">{}</span>'.format(' '.join(classes), text)


def table(clazz='border mt-3',
          id_='',
          hcols=None,
          bcols=None,
          fcols=None,
          head=None,
          body=None,
          foot=None):
    return {
        'clazz': clazz,
        'id': id_,
        'hcols': hcols,
        'bcols': bcols,
        'fcols': fcols,
        'head': head,
        'body': body,
        'foot': foot
    }
