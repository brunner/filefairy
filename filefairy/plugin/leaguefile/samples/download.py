#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import re
import sys

_path = os.path.dirname(os.path.abspath(__file__))
_root = re.sub(r'/plugin/leaguefile/samples', '', _path)
sys.path.append(_root)
from util.component.component import card  # noqa
from util.component.component import table  # noqa

_breadcrumbs = [{
    'href': '/fairylab/',
    'name': 'Home'
}, {
    'href': '',
    'name': 'Leaguefile'
}]

_cols = [' class="w-55p"', '']
_table = table(
    clazz='table-sm',
    hcols=_cols,
    bcols=_cols,
    body=[['Time: ', '4h 56m'], ['Size: ', '359,969,530']])
_upload = card(title='Mar 10', table=_table, ts='3m ago', success='completed')

_table = table(
    clazz='table-sm',
    hcols=_cols,
    bcols=_cols,
    body=[['Time: ', '3m'], ['Size: ', '59,969,530']])
_download = card(title='Mar 10', table=_table, ts='0s ago', success='ongoing')

_ccols = [
    '', ' class="text-center"', ' class="text-center"', ' class="text-right"'
]
_completed = table(
    clazz='border mt-3',
    hcols=_ccols,
    bcols=_ccols,
    head=['Date', 'Upload', 'Download', 'Size'],
    body=[['Mar 8', '10h 11m', '11m', '358,347,534'],
          ['Mar 6', '9h 34m', '6m', '356,922,996']])

subtitle = ''

tmpl = 'leaguefile.html'

context = {
    'title': 'leaguefile',
    'breadcrumbs': _breadcrumbs,
    'upload': _upload,
    'download': _download,
    'completed': _completed
}
