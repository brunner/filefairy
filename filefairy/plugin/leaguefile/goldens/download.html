<!doctype html>
<html lang="en">

<head>

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link rel="stylesheet"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
      integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
      crossorigin="anonymous">
<link rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/open-iconic/1.1.1/font/css/open-iconic-bootstrap.min.css">
<title>Fairylab | leaguefile</title>

<style>
html {
  position: relative;
  min-height: 100%;
}
body {
  margin-bottom: 60px;
}
header {
  background-color: #f5f5f5;
  width: 100%;
}
footer {
  background-color: #f5f5f5;
  bottom: 0;
  height: 60px;
  line-height: 60px;
  position: absolute;
  width: 100%;
}
.alert, .form-control {
  padding: .75rem;
}
.alert .oi {
  line-height: 24px;
  padding-right: .75rem;
}
.anchor {
  padding-top: 0px;
  margin-top: 0px;
  transition: padding-top .35s ease, margin-top .35s ease;
}
.breadcrumb {
  background-color: inherit;
  line-height: 60px;
}
.flex-grow {
  flex-grow: 1;
}
.left-8p {
  left: 8px;
}
.navbar {
  padding: 0;
}
.navbar-nav {
  padding: .5rem 0;
}
.navbar .nav-link {
  padding: .5rem;
}
.pl-24p {
  padding-left: 24px;
}
.pr-24p {
  padding-right: 24px;
}
.right-8p {
  right: 8px;
}
.table, .table-sm {
  margin-bottom: 0;
  word-break: break-word;
}
.table .thead-light td {
  background-color: #f5f5f5;
}
.table-fixed {
  table-layout: fixed;
}
.table-sm td {
  border-top: 0;
  padding: 0 .3rem 0 0;
}
.table .td-sm {
  padding: .75rem .25rem;
}
.top-14p {
  top: 14px;
}
.w-10 {
  width: 10%;
}
.w-20 {
  width: 20%;
}
.w-40 {
  width: 40%;
}
.w-55p {
  width: 55px;
}
.w-65p {
  width: 65px;
}
.w-75p {
  width: 75px;
}
.w-105p {
  width: 105px;
}
</style>
</head>

<body>

<!-- ^(base.header) -->
<header class="border-bottom sticky-top">
<div class="container">
<nav class="navbar navbar-light" aria-label="nav">
<ol class="breadcrumb m-0 p-0">
<li class="breadcrumb-item">
<a href="/">Fairylab</a>
</li>
<li class="breadcrumb-item active" aria-current="page">Leaguefile</li>
</ol>

<!-- ^(common.nav) -->
<button class="navbar-toggler border-0" type="button" data-toggle="collapse"
        data-target="#nav-dropdown" aria-controls="#nav-dropdown"
        aria-expanded="false" aria-label="nav toggler">
<span class="navbar-toggler-icon"></span>
</button>
<div class="collapse navbar-collapse border-top" id="nav-dropdown">
<ul class="navbar-nav">
<li class="nav-item">
<a class="nav-link" href="#upload">Upload</a>
</li>
<li class="nav-item">
<a class="nav-link" href="#download">Download</a>
</li>
<li class="nav-item">
<a class="nav-link" href="#completed">Completed</a>
</li>
</ul>
</div>
<!-- $(common.nav) -->

</nav>
</div>
</header>
<!-- $(base.header) -->

<!-- ^(base.main) -->
<main role="main" class="pb-3">

<!-- ^(leaguefile.upload) -->
<div class="container anchor" id="upload">

<!-- ^(common.heading) -->
<div class="row">
<div class="col-12">
<div class="alert alert-secondary mt-3 mb-0">
<span class="oi oi-cloud-upload text-secondary"></span>Upload
</div>
</div>
</div>
<!-- $(common.heading) -->

<!-- ^(leaguefile.row) -->
<div class="row">

<!-- ^(leaguefile.column) -->
<div class="col-12">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<div class="d-flex flex-row">
<h5 class="card-title flex-grow">
Mar 10
</h5>
<div><span class="btn btn-success btn-sm disabled">completed</span></div>
</div>


<!-- ^(common.table) -->
<table class="table table-sm">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="w-55p">Time: </td>
<td>4h 56m</td>
</tr>
<tr>
<td scope="row" class="w-55p">Size: </td>
<td>359,969,530</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 06:02:30 EDT (1985-10-26)</small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- $(leaguefile.column) -->

</div>
<!-- ^(leaguefile.row) -->

</div>
<!-- $(leaguefile.upload) -->

<!-- ^(leaguefile.download) -->
<div class="container anchor" id="download">

<!-- ^(common.heading) -->
<div class="row">
<div class="col-12">
<div class="alert alert-secondary mt-3 mb-0">
<span class="oi oi-cloud-download text-secondary"></span>Download
</div>
</div>
</div>
<!-- $(common.heading) -->

<!-- ^(leaguefile.row) -->
<div class="row">

<!-- ^(leaguefile.column) -->
<div class="col-12">

<!-- ^(common.card) -->
<div class="card d-flex flex-column flex-grow mt-3">

<!-- ^(common.card.body) -->
<div class="card-body d-flex flex-column">
<div class="d-flex flex-row">
<h5 class="card-title flex-grow">
Mar 10
</h5>
<div><span class="btn btn-success btn-sm disabled">ongoing</span></div>
</div>


<!-- ^(common.table) -->
<table class="table table-sm">

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row" class="w-55p">Time: </td>
<td>3m</td>
</tr>
<tr>
<td scope="row" class="w-55p">Size: </td>
<td>59,969,530</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- $(common.card.body) -->

<!-- ^(common.card.footer) -->
<div class="card-footer text-muted text-truncate">
<small>Last activity: 06:02:30 EDT (1985-10-26)</small>
</div>
<!-- $(common.card.footer) -->

</div>
<!-- $(common.card) -->

</div>
<!-- ^(leaguefile.column) -->

</div>
<!-- ^(leaguefile.row) -->

</div>
<!-- $(leaguefile.upload) -->

<!-- ^(leaguefile.completed) -->
<div class="container anchor" id="completed">

<!-- ^(common.heading) -->
<div class="row">
<div class="col-12">
<div class="alert alert-secondary mt-3 mb-0">
<span class="oi oi-task text-secondary"></span>Completed
</div>
</div>
</div>
<!-- $(common.heading) -->

<!-- ^(leaguefile.row) -->
<div class="row">

<!-- ^(leaguefile.column) -->
<div class="col-12">

<!-- ^(common.table) -->
<table class="table border mt-3">

<!-- ^(common.table.head) -->
<thead class="thead-light">
<tr>
<td scope="col">Date</td>
<td scope="col" class="text-center">Upload</td>
<td scope="col" class="text-center">Download</td>
<td scope="col" class="text-right">Size</td>
</tr>
</thead>
<!-- $(common.table.head) -->

<!-- ^(common.table.body) -->
<tbody>
<tr>
<td scope="row">Mar 8</td>
<td class="text-center">10h 11m</td>
<td class="text-center">11m</td>
<td class="text-right">358,347,534</td>
</tr>
<tr>
<td scope="row">Mar 6</td>
<td class="text-center">9h 34m</td>
<td class="text-center">6m</td>
<td class="text-right">356,922,996</td>
</tr>
</tbody>
<!-- $(common.table.body) -->

</table>
<!-- $(common.table) -->

</div>
<!-- ^(leaguefile.column) -->

</div>
<!-- ^(leaguefile.row) -->

</div>
<!-- $(leaguefile.upload) -->

</main>
<!-- $(base.main) -->

<!-- ^(base.footer) -->
<footer class="border-top">
<div class="container text-center">
<small class="text-muted">Last reloaded: 06:02:30 PDT (1985-10-26)</small>
</div>
</footer>
<!-- $(base.footer) -->

<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
        integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
        crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
        integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
        crossorigin="anonymous"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js"
        integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4"
        crossorigin="anonymous"></script>
<script>
$('.navbar-toggler').one('click', function(e) {
  setTimeout(function() {
    var height = $('header').height() + 1;
    var css = '<style> .anchor.offset { padding-top: %spx; margin-top: -%spx; } </style>';
    $(css.replace(/%s/g, height)).appendTo('head');
  }, 500);
});
$('.navbar-toggler').on('click', function(e) {
  $('.anchor').toggleClass('offset');
});
</script>


</body>

</html>